aten/src/ATen/native/cuda/Sort.cu

```diff
@@ -85,6 +265,33 @@ std::tuple<Tensor &,Tensor &> sort_out_stable_cuda(const Tensor & self, c10::opt
 
   TORCH_CHECK(nsort <= std::numeric_limits<int>::max(),
     "The dimension being sorted can not have more than INT_MAX elsments.");
+  // use inplace algorithm for smaller input sizes without stable=True
+  if (should_use_small_sort(self, dim) && !stable.value()) {
+    // from thc: sorted->values, indices->indices, input->self
+    TensorArg self_arg{self, "self", 1}, values_arg{values, "values", 2}, indices_arg{indices, "indices", 3};
+    checkAllSameGPU("small_sort", {self_arg, values_arg, indices_arg});
+    at::maybe_wrap_dim(dim, self);
+
+    if (!values.defined()) {
+      values = at::empty_like(self, self.options());
+    }
+    if (!indices.defined()) {
+      indices = at::empty_like(self, self.options().dtype(kLong));
+    }
```

Comment:
Would this preserve the memory format of self?

Reply:
I don't know if it is guaranteed, but it does not break the stride tests...

Reply:
Yes, *like functions preserve memory format of self (to the best extent possible, if e.g. self is broadcasted tensor with 0 strides then the result is still memory dense, but I don't remember where the broadcasted dimension would physically go).